bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/core/GraphqlProductViewHandler.java

```diff
@@ -108,6 +140,7 @@ private static Hit createHit(Product product, XSSAPI xssAPI) {
         hit.set("title", xssAPI.encodeForHTML(title));
         hit.set("sku", xssAPI.encodeForHTML(product.getSKU()));
         hit.set("excerpt", xssAPI.filterHTML(""));
+        hit.set("resource", resource);
```

Comment:
What is the idea behind the change this to resource? Also if you only use the resource in line 116ff you don't need to set other product attributes individually if they are not used?

Reply:
The idea is to reuse the resource returned by the search. It's used in generateHtmlOutput() to avoid resolving the resource again which generates a new GraphQL request for each particular search hit.

Reply:
@mhaack , the search hits have also a JSON rendering, the product attributes are still needed for that.

Reply:
That is right, but the JSON rendering only converts to hit to a JSON object with key/value pairs. So we could remove line 340 `Product product = resource.adaptTo(Product.class);` and change create hit to `createHit(resource, xssAPI)`. This will require 1 or 2 lines more to set the sku or title for the `hit` object removes one more adaptTo call.

Reply:
@mhaack , what do you want to achieve with that? An adaption to product is not a performance issue and product is also used in the card item renderer, backed by MagentProduct which encapsulates logic for retrieving the thumbnail. 
I would prefer keeping this PR for the topic of fixing those more serious performance issues related to XL catalogs.

Reply:
It seems duplicate to me. If the resource is used only, as you wrote the rendering does a resource.adaptTo(Product.class). Why would we need the individual fields here? 
We can keep if for this PR and address later, I'm ok with that.

Reply:
If you look in the code they were apparently duplicate earlier as well, because when the results are rendered the data is taken from the product not from the hit. 
So if I keep the existing data in the hit then it's very likely I don't break anything, while removing data from it might affect the JSON rending of the hit, wherever is that used.